### PR TITLE
Improve accessability of Span<T> methods.

### LINKED ISSUE #1164: Improve accessability of Span<T> methods on Image<T> / ImageFrame<T>
## Problems

We have two, somewhat related issues to address:
1. `AdvancedImageExtensions.GetPixelSpan` is now obsolete, and there is no convenient way to get a single pixel span for memory interop use-cases
2. Keeping the pixel span methods in `AdvancedImageExtensions` leads to poor discoverability. With all the Roslyn analyzers we have today, to detect `Span<T>` misuse, I no longer consider them unsafe or "advanced". I suggest to make them instance members on `Image<TPixel>` and `ImageFrame<TPixel>`. Note: this does not apply for `Memory<T>` or `IMemoryGroup<T>` stuff, since persisting heap-references to the pixel memory has some nontrivial consequences.

## Proposal

Redifine `Span<TPixel>` methods as following:
```C#
public class Image<TPixel>
{
    public bool TryGetSinglePixelSpan(out Span<TPixel> pixelSpan);
    public Span<TPixel> GetPixelRowSpan(int rowIndex);
}

public class ImageFrame<TPixel>
{
    public bool TryGetSinglePixelSpan(out Span<TPixel> span);
    public Span<TPixel> GetPixelRowSpan(int rowIndex);       
}
```

**Update:** Fixed copy/paste errors in the proposal.

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Fixes #1164

### CODE REVIEW

File: src/ImageSharp/ImageFrame{TPixel}.cs [hunk 1]
```diff
@@ -166,6 +167,39 @@ internal ImageFrame(Configuration configuration, ImageFrame<TPixel> source)
...
+        /// </summary>
+        /// <param name="span">The <see cref="Span{T}"/>.</param>
+        /// <returns>The <see cref="bool"/>.</returns>
+        public bool TryGetSinglePixelSpan(out Span<TPixel> span)
+        {
+            IMemoryGroup<TPixel> mg = this.GetPixelMemoryGroup();
+            if (mg.Count > 1)
+            {
+                span = default;
+                return false;
```
Comment (JimBobSquarePants) [inline src/ImageSharp/ImageFrame{TPixel}.cs:196]: @antonfirsov Is there a test somewhere I can tap into to test the false condition?

Comment (antonfirsov) [reply src/ImageSharp/ImageFrame{TPixel}.cs:196]: Yes, you can limit `MemoryAllocator`'s `BufferCapacity` in many ways to ensure the backing memory is fragmented: 
- [like this](https://github.com/SixLabors/ImageSharp/blob/master/tests/ImageSharp.Tests/Memory/Buffer2DTests.cs#L43) (works with `TestMemoryAllocator` only).
- [or like this](https://github.com/SixLabors/ImageSharp/blob/master/tests/ImageSharp.Tests/Formats/Jpg/JpegDecoderTests.cs#L109).

File: src/ImageSharp/ImageFrame{TPixel}.cs [hunk 2]
```diff
@@ -166,6 +167,39 @@ internal ImageFrame(Configuration configuration, ImageFrame<TPixel> source)
             }
         }
 
+        /// <summary>
+        /// Gets the representation of the pixels as a <see cref="Span{T}"/> of contiguous memory
+        /// at row <paramref name="rowIndex"/> beginning from the first pixel on that row.
+        /// </summary>
+        /// <param name="rowIndex">The row.</param>
+        /// <returns>The <see cref="Span{TPixel}"/></returns>
```
Comment (antonfirsov) [inline src/ImageSharp/ImageFrame{TPixel}.cs:175]: The docs are missing `<exception>` for `ArgumentOutOfRangeException`.

File: src/ImageSharp/Image{TPixel}.cs [hunk 3]
```diff
@@ -163,6 +163,39 @@ internal Image(Configuration configuration, ImageMetadata metadata, IEnumerable<
             }
         }
 
+        /// <summary>
+        /// Gets the representation of the pixels as a <see cref="Span{T}"/> of contiguous memory
+        /// at row <paramref name="rowIndex"/> beginning from the first pixel on that row.
+        /// </summary>
+        /// <param name="rowIndex">The row.</param>
+        /// <returns>The <see cref="Span{TPixel}"/></returns>
+        public Span<TPixel> GetPixelRowSpan(int rowIndex)
```
Comment (antonfirsov) [inline src/ImageSharp/Image{TPixel}.cs:172]: See previous note.

**Pull request merged**